src/models/db/subscriptions.js

```diff
@@ -48,7 +55,11 @@ function * upsertSubscription (subscription, options) {
 }
 
 function * deleteSubscription (id, options) {
-  yield Subscription.destroy(_.assign({}, options, {where: {id: id}}))
+  const existingSubscription = yield getSubscription(id, options)
```

Comment:
I think it is: `db.deleteSubscription(id, {transaction})`. As long as the transaction is passed in, it should be used throughout this method, right? I can add an `assert(options.transaction)` as in `_upsertSubscription`